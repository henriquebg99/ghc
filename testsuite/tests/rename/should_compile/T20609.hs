module T20609 where

-- Triggers the warning (definition/binding sites):
-- ------------------------------------------------

forall x = ()

(âˆ€) x = ()

fparam forall = ()

asPattern forall@Nothing = ()

localLet = let forall = () in forall

{-# RULES "rule" forall forall. id forall = forall #-}

{-# RULES "rule_sig" forall a. forall (forall :: a). id forall = forall #-}

-- Does not trigger the warning (use sites):
-- -----------------------------------------

other = forall
