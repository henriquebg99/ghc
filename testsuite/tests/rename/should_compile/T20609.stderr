
T20609.hs:6:1: warning: [-Wforall-identifier (in -Wdefault)]
    The use of ‘forall’ as an identifier
    will become an error in a future GHC release.
    Suggested fix:
      Consider using another name, such as
      ‘forAll’, ‘for_all’, or ‘forall_’.

T20609.hs:8:1: warning: [-Wforall-identifier (in -Wdefault)]
    The use of ‘∀’ as an identifier
    will become an error in a future GHC release.
    Suggested fix:
      Consider using another name, such as
      ‘forAll’, ‘for_all’, or ‘forall_’.

T20609.hs:10:8: warning: [-Wforall-identifier (in -Wdefault)]
    The use of ‘forall’ as an identifier
    will become an error in a future GHC release.
    Suggested fix:
      Consider using another name, such as
      ‘forAll’, ‘for_all’, or ‘forall_’.

T20609.hs:12:11: warning: [-Wforall-identifier (in -Wdefault)]
    The use of ‘forall’ as an identifier
    will become an error in a future GHC release.
    Suggested fix:
      Consider using another name, such as
      ‘forAll’, ‘for_all’, or ‘forall_’.

T20609.hs:14:16: warning: [-Wforall-identifier (in -Wdefault)]
    The use of ‘forall’ as an identifier
    will become an error in a future GHC release.
    Suggested fix:
      Consider using another name, such as
      ‘forAll’, ‘for_all’, or ‘forall_’.

T20609.hs:16:25: warning: [-Wforall-identifier (in -Wdefault)]
    The use of ‘forall’ as an identifier
    will become an error in a future GHC release.
    Suggested fix:
      Consider using another name, such as
      ‘forAll’, ‘for_all’, or ‘forall_’.

T20609.hs:18:40: warning: [-Wforall-identifier (in -Wdefault)]
    The use of ‘forall’ as an identifier
    will become an error in a future GHC release.
    Suggested fix:
      Consider using another name, such as
      ‘forAll’, ‘for_all’, or ‘forall_’.
